JuliaLowering/src/macro_expansion.jl

```diff
@@ -454,6 +453,9 @@ function expand_forms_1(ctx::MacroExpansionContext, ex::SyntaxTree)
         end
     elseif k == K"macrocall"
         expand_macro(ctx, ex)
+    elseif k == K"toplevel" && length(ctx.scope_layer_stack) > 1
+        fix_toplevel_expansion(ctx, ex, current_layer(ctx).mod,
+                               source_location(LineNumberNode, ex))
```

Comment:
This seems to change the LineNumberNode slightly (allows `file === nothing`), but I guess we don't use this in lowering right now.

Does this affect anything?